lib/Twig/Environment.php

```diff
@@ -107,6 +108,8 @@ public function __construct(Twig_LoaderInterface $loader = null, $options = arra
             'optimizations' => -1,
         ), $options);
 
+        // cache can be object so remove it from options hash
+        $this->optionsHash = serialize(array_diff_key($options, array('cache' => null)));
```

Comment:
There is no property for an `optimization` option, for example. But template differs when change this option. I see two approaches:

a. add `options` property with all options and use it instead of `debug`/`autoReload` 
b. add property for optimization or create getter `Twig_Extension_Optimizer::getOptimizattion` and call it every time